### PR TITLE
Setup SW Architecture with Structurizr DSL

### PR DESCRIPTION
**Pull request by Andrei-Basarab**
For more context, visit:
- https://c4model.com/
- https://structurizr.com/dsl
- https://structurizr.com/help/lite
- https://github.com/structurizr/dsl

### CODE REVIEW

**Review by ciprian-barbu — DISMISSED** [review]
Looks good to me, but I don't have Docker installed in my Windows environment; I will test on Linux as soon as possible

File: README.md [hunk 1]
```diff
@@ -1,67 +1,90 @@
...
+
+#### Prerequisites
+Finish [Basic](#basic) setup
+
+#### Scope
+Design, Development, Testing, Debugging, etc.
+
+#### Steps
+- [Git](https://git-scm.com/downloads)
+- [VSCode](https://code.visualstudio.com/)
```
Comment (ciprian-barbu) [inline README.md]: The documentation looks good now, but at least for this VSCode part you might want to highlight it is for Windows only.
Generally speaking, I think we need complete but separate guides (from beginning to end) for Windows and Linux environments, but this is a task by itself.

Comment (DanDragan) [reply README.md]: VsCode is available for Linux environments, too. But, I also support the fact that we should not be bound by a specific IDE. Maybe we can put in place a package a dependency and environment management (package management) like Conda or Conan and create multiple environments customized to smaller groups of people.

**Review by DanDragan — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (Andrei-Basarab) [reply README.md]: As @DanDragan said, VSCode indeed is cross-OS compatible, but of course it is not mandatory but rather recommended (I will add a mention about it).
Regarding separate guidelines for each OS, we had it like this previously, but I decided to go into a more hybrid approach.
First I separated Setup into `Basic` and `Complete` chapters and then listed all necessary tools that go in each chapter. For some tools, there might be OS specific setup.
@DanDragan, regarding the environment and package management, maybe we can try it in the future in a separate task, based on its necessity and added value.

Comment (ciprian-barbu) [reply README.md]: Ok, I didn't think to search if VSCode works on Linux.
At this stage, the instructions guide can actually work on a more traditional, shell based environment, I've checked this by running in MSYS2 and partially on cygwin. I also intend to run on my Linux box.
However, I found an issue with arm-none-eabi-gdb, which needs to be documented as a known problem for at least MSYS2.

**Pull request merged**